File: frontend/src/types/deal.ts
/** AI 优惠（与后端 DealResponse 对齐） */
export type DealStatus = 'DRAFT' | 'ACTIVE' | 'ENDED' | 'OFFLINE'

export type DealDto = {
  id: number
  title: string
  description: string
  promoCode: string | null
  /** 活动链接；空则前台不展示 */
  url: string | null
  startsAt: string
  endsAt: string
  status: DealStatus
  toolId: number | null
  toolSlug: string | null
  toolName: string | null
  /** 关联产品 Logo（来自 tool.logo_url） */
  toolLogoUrl?: string | null
  createdAt?: string
  updatedAt?: string
}

export type DealUpsertBody = {
  title: string
  description: string
  promoCode?: string | null
  url?: string | null
  startsAt: string
  endsAt: string
  status: DealStatus
  /** 必填：关联 AI 产品（卡片展示其 Logo） */
  toolId: number
}

File: backend/src/main/java/com/zzh/personal_hub/deal/entity/Deal.java
package com.zzh.personal_hub.deal.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "deal")
public class Deal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tool_id")
    private Long toolId;

    @Column(nullable = false, length = 120)
    private String title;

    @Column(nullable = false, length = 1000)
    private String description;

    @Column(name = "promo_code", length = 64)
    private String promoCode;

    @Column(length = 500)
    private String url;

    @Column(name = "starts_at", nullable = false)
    private Instant startsAt;

    @Column(name = "ends_at", nullable = false)
    private Instant endsAt;

    /** DRAFT | ACTIVE | ENDED | OFFLINE */
    @Column(nullable = false, length = 16)
    private String status = "DRAFT";

    @Column(nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    @Column(nullable = false)
    private Instant updatedAt = Instant.now();

    @Column(name = "deleted_at")
    private Instant deletedAt;
}
